### DISCUSSION TITLE
Unable to Export Traces to Jaeger

### DISCUSSION THREAD

**Discussion by DaGeRe** [jaegertracing/jaeger#4283] [Q&A] [answered]
I'm just getting started to export traces (which, in the future, should transform existing Kieker traces into OpenTelemetry traces). To do so, I just started a jaeger server (as described in https://www.jaegertracing.io/docs/1.6/getting-started/ and https://medium.com/jaegertracing/introducing-native-support-for-opentelemetry-in-jaeger-eb661be8183c):
```
docker run -d --name jaeger \
  -p 16686:16686 \
  -e COLLECTOR_OTLP_ENABLED=true \
  -p 4318:4318 \
  jaegertracing/all-in-one:1.42
```

and tried to create spans with a very simple Java program:

```
public class MySpanCreator {

	private static final String CREATOR_NAME = "test";
	private static final String CREATOR_VERSION = "2.0.0-SNAPSHOT";

	private final Tracer tracer;

	public static void main(String[] args) throws InterruptedException {
		MySpanCreator creator = new MySpanCreator();
		for (int i = 0; i < 100; i++) {
			creator.createSpan("MySpanName" + i);
		}
		Thread.sleep(1000);
	}

	public MySpanCreator() {
		Resource resource = Resource.getDefault()
				.merge(Resource.create(Attributes.of(ResourceAttributes.SERVICE_NAME, "createdSpans")));
		
		OtlpGrpcSpanExporter spanExporter = OtlpGrpcSpanExporter.builder()
				.setEndpoint("http://127.0.0.1:4318")
				.build();

		SdkTracerProvider sdkTracerProvider = SdkTracerProvider.builder()
				.addSpanProcessor(BatchSpanProcessor.builder(spanExporter)
						.setScheduleDelay(100, TimeUnit.MILLISECONDS).build())
				.setResource(resource)
				.build();

		OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
				.setTracerProvider(sdkTracerProvider)
				.buildAndRegisterGlobal();
		
		tracer = openTelemetry.getTracer(CREATOR_NAME, CREATOR_VERSION);
	}

	public void createSpan(String name) {
		SpanBuilder spanBuilder = tracer.spanBuilder(name);
		spanBuilder.setStartTimestamp(System.currentTimeMillis(), TimeUnit.MILLISECONDS);
		Span span = spanBuilder.startSpan();
		span.end(System.currentTimeMillis() + 15, TimeUnit.MILLISECONDS);


		System.out.println("Creating span: " + name);
		try (Scope ss = span.makeCurrent()) {
			// Do nothing, since we alreay set start and end time
		} catch (Throwable t) {
			t.printStackTrace();
		} finally {
			span.end();
		}
	}
}
```
as described in https://opentelemetry.io/docs/instrumentation/java/manual/ (and with some hints from https://www.logicmonitor.com/blog/java-application-manual-instrumentation-for-distributed-traces).

The jaeger UI shows up, but I'm not able to get any traces, besides the jaeger-traces queries. The log shows

```
Mär 08, 2023 11:52:23 AM io.opentelemetry.sdk.internal.ThrottlingLogger doLog
SCHWERWIEGEND: Failed to export spans. The request could not be executed. Full error message: null
```
Port 4318 is reachable, at least according to the jaeger log:

`{"level":"info","ts":1678269522.6670544,"caller":"otlpreceiver@v0.70.0/otlp.go:112","msg":"Starting HTTP server","endpoint":"0.0.0.0:4318"}`

Also with changing details of the implementation, I was not able to get any traces into jaeger. Could anybody give me an hint what is missing here?

**Comment by albertteoh** [comment, marked as answer]
It looks like you're using an OTLP gRPC exporter. Have you tried connecting to port `4317`? That's the port that accepts OTLP over gRPC: https://www.jaegertracing.io/docs/1.42/getting-started/#all-in-one.

**Reply by DaGeRe** [reply]
Thanks a lot, using port 4317 solved the issue!
